### PR TITLE
Add end to end tests to show correct detection of mis-downloaded files

### PR DESCRIPTION
**Pull request by directionless**
This adds some tests that show correct detection of incorrectly downloaded files. 

It has some light refactoring to fit with my understanding of best practices.

### CODE REVIEW

File: tuf/delegation_test.go [hunk 1]
```diff
@@ -397,12 +442,77 @@ func TestEndToEnd(t *testing.T) {
 		{"nil autoupdate func", 1, 2, wontCrashOnNilAutoupdate},
 		{"autoupdate interval works", 1, 2, autoupdateDetectedChangeAfterInterval},
 		{"interleaved operations", 1, 2, interleavedOperations},
+		{"truncated download", 1, 2, genCorruptDownloadTest(replaceBodyCorruption, errLengthIncorrect)},
+		{"corrupt download", 1, 2, genCorruptDownloadTest(overwriteCorruption, errHashIncorrect)},
```
Comment (blaedj) [inline tuf/delegation_test.go]:
```suggestion
		{"corrupt download", 1, 2, genCorruptDownloadTest(overwriteCorruption, errHashIncorrect)},
		{"empty download", 1, 2, genCorruptDownloadTest(emptyBodyCorruption, errLengthIncorrect)},

```

**Review by blaedj — APPROVED** [review]
lgtm, one suggestion to add a test case for empty request bodies, since the functionality is there in the `genCorrupDownloadTest` anyway
